semgrep-core/matching/Matching_generic.ml

```diff
@@ -218,16 +218,22 @@ let equal_ast_binded_code (a: AST.any) (b: AST.any) : bool =
        *)
       let a = Lib.abstract_position_info_any a in
       let b = Lib.abstract_position_info_any b in
-      let res = a =*= b in
-      if !Flag.debug_matching && not res
-      then begin
-        pr2 (spf "A = %s" (str_of_any a));
-        pr2 (spf "B = %s" (str_of_any b));
-      end;
-      res
-
+      a =*= b
+  | A.I _, A.E (A.Id (b_id, _)) ->
+    (* Allow identifier nodes to match pure identifier expressions *)
+      equal_ast_binded_code a (A.I b_id)
   | _, _ ->
       false
+  ) in
+
+  if !Flag.debug_matching && not res
+  then begin
```

Comment:
Note that I also print debugging information even when the two node types differ.